test/over_react/component/error_boundary_test.dart

```diff
@@ -34,6 +38,64 @@ void main() {
     });
 
     // TODO: Add tests that exercise the actual ReactJS 16 error lifecycle methods once implemented.
+    group('catches component errors', () {
+      List<Map<String, List>> calls;
+      DivElement mountNode;
+
+      void verifyReact16ErrorHandlingWithoutErrorBoundary() {
+        mountNode = new DivElement();
+        document.body.append(mountNode);
+        var jacketOfFlawedComponentWithNoErrorBoundary = mount(Flawed()(), mountNode: mountNode);
+        expect(mountNode.children, isNotEmpty, reason: 'test setup sanity check');
+        jacketOfFlawedComponentWithNoErrorBoundary.getNode().click();
+        expect(mountNode.children, isEmpty,
+            reason: 'rendered trees not wrapped in an ErrorBoundary '
+                    'should get unmounted when an error is thrown within child component lifecycle methods');
+
+        mountNode.remove();
+        mountNode = new DivElement();
+        document.body.append(mountNode);
+      }
+
+      setUp(() {
+        // Verify the behavior of a component that throws when it is not wrapped in an error boundary first
+        verifyReact16ErrorHandlingWithoutErrorBoundary();
+
+        calls = [];
+        jacket = mount(
+          (ErrorBoundary()
+            ..onComponentDidCatch = (err, info) {
+              print('onComponentDidCatch');
+              calls.add({'onComponentDidCatch': [err, info]});
+            }
+          )(Flawed()()),
+          mountNode: mountNode,
+        );
+        expect(mountNode.children, isNotEmpty, reason: 'test setup sanity check');
+        // Cause an error to be thrown within a ReactJS lifecycle method
+        jacket.getNode().click();
+      });
+
+      tearDown(() {
+        mountNode.remove();
+        mountNode = null;
+      });
+
+      test('and calls `props.onComponentDidCatch`', () {
+        expect(calls.single.keys, ['onComponentDidCatch']);
+        final errArg = calls.single['onComponentDidCatch'][0];
+        expect(errArg.toString(), contains('FlawedComponent.componentWillUpdate'));
```

Comment:
Can we also verify that this is of the correct type, and that the error has been successfully Dart-ified?

```suggestion
        expect(errArg, const isInstanceOf<Error>());
        expect(errArg.toString(), contains('FlawedComponent.componentWillUpdate'));
```